src/main/java/androidx/text/SpannableString.kt

```diff
@@ -70,3 +70,59 @@ inline operator fun Spannable.set(range: IntRange, span: Any) {
     // This looks weird, but endInclusive is just the exact upper value.
     setSpan(span, range.start, range.endInclusive, SPAN_INCLUSIVE_EXCLUSIVE)
 }
+
+/**
+ * Apply a [span] to the [substring] in the text.
+ *
+ * ```
+ * val s = "Hello, World!".toSpannable()
+ * s.overlay("Hello", UnderlineSpan())
+ * ```
+ *
+ * Note: Does nothing if the substring is not found.
+ *
+ * @param ignoreCase true to ignore character case when matching a string. By default false.
+ * @param startIndex starting index to search the string. Default: 0
+ *
+ * @return the end index (exclusive) of the applied span, or -1 if not found.
+ *
+ * @see Spannable.setSpan
+ */
+fun Spannable.overlay(
+    substring: String,
+    span: Any,
+    ignoreCase: Boolean = false,
+    startIndex: Int = 0
+): Int {
+    val start = indexOf(substring, startIndex, ignoreCase)
+    if (start == -1) return start
+    val end = start + substring.length
+    setSpan(span, start, end, Spannable.SPAN_INCLUSIVE_INCLUSIVE)
+    return end
+}
+
+/**
+ * Apply a [span] to all occurrences of [substring] in the text.
+ *
+ * ```
+ * val s = "Hello, World! I'm World".toSpannable()
+ * s.overlayAll("World", UnderlineSpan())
+ * ```
+ *
+ * Note: Does nothing if the substring is not found.
+ *
+ * @param ignoreCase true to ignore character case when matching a string. By default false.
+ * @param startIndex starting index to search the string. Default: 0
+ *
+ * @see Spannable.setSpan
+ */
+inline fun Spannable.overlayAll(
```

Comment:
I don't think this should be `inline` as it's not a simple alias of one or two function calls.

Reply:
Wasn't sure the rule about using `inline`. Seemed like everything was using it, so I just stuck with it where it didn't cause problems. Just a regular `fun` now.